public/scripts/common/lib/handlebars.mixed.js

```diff
@@ -29,10 +29,53 @@ Displays select tag with options (and optional optgroups)
   * `selected`
 
 **/
+
 Handlebars.registerHelper('select', function(context, options) {
-  var output = `<select id="${options.hash.id}" name="${options.hash.name}" class="${options.hash.class}">`;
-  output += makeOptions(context, options);
-  return output +'</select>';
+  var addOption, addOptions,
+      output = '';
+
+  addOption = function(selected, _options) {
+    var {value, name, disabled, className, isSelected} = _options;
+    if(_.isNull(value)) value = 'null-' + _.uniqueId();
+    
+    if(_.isNull(isSelected)) {
+      isSelected = value == selected;
+    }
+
+
+    return  '<option' + 
+            (value ? ` value="${value}"` : '') +
+            (isSelected ? ' selected="selected"' : '') +
+            (disabled ? ' disabled="disabled"' : '') +
+            (className ? ` class="${className}"` : '') +
+            '>' + name + '</option>';
+  };
+
+  addOptions = function(selected, _options) {
+    return _.map(_options, _.partial(addOption, selected)).join('');
+  };
+
+  var requiredString = "";
+  if(options.hash.required) {
+    requiredString = " required ";
+  }
+
+  output += '<select id="' + options.hash.id + '" name="' +
+            options.hash.name + '" class="' + options.hash.class + '"' + requiredString + '>';
```

Comment:
I like the idea, but setting the `select` tag to be `required` doesn't seem to work when an option is already selected.  i.e. if you fill out the short name for this part and then hit "Create new RDP part", it errors with `TypeError: Invalid partType: "null"`.  Not sure if there's a nice solution to this or not.